packages/ember-glimmer/lib/utils/iterable.js

```diff
@@ -68,19 +82,34 @@ class EmberArrayIterator {
     this.length = get(array, 'length');
     this.keyFor = keyFor;
     this.position = 0;
+    this.guid = null;
+    this.seen = {};
   }
 
   isEmpty() {
     return this.length === 0;
   }
 
   next() {
-    let { array, length, keyFor, position } = this;
+    let { array, length, keyFor, position, guid, seen } = this;
 
     if (position >= length) { return null; }
 
     let value = objectAt(array, position);
     let key = keyFor(value, position);
+    let hasSeen = seen[key];
+
+    if (hasSeen) {
+      if (!guid) {
+        this.guid = guidFor(array);
```

Comment:
Is the idea to brand it like how things are branded in Glimmer e.g. [this](https://github.com/tildeio/glimmer/blob/master/packages/glimmer-reference/lib/const.ts#L5).